api/turing/cluster/servicebuilder/service_builder.go

```diff
@@ -414,6 +404,42 @@ func (sb *clusterSvcBuilder) getTopologySpreadConstraints() ([]corev1.TopologySp
 	return topologySpreadConstraints, nil
 }
 
+func (sb *clusterSvcBuilder) getCPULimit(resourceRequest *models.ResourceRequest) *resource.Quantity {
```

Comment:
do you think this would be easier to read 
```
if resourceRequest == nil {
    return nil
}

# use user set limit as priority
if resourceRequest.CPULimit != nil && !resourceRequest.CPULimit.IsZero() {
    return resourceRequest.CPULimit
}

if sb.knativeServiceConfig.UserContainerCPULimitRequestFactor == 0 {
    return nil
}

# apply platform multiplier if limit is not set
cpuLimit := cluster.ComputeResource(resourceRequest.CPURequest, sb.knativeServiceConfig.UserContainerCPULimitRequestFactor)
return &cpuLimit
```